File: apps/android/app/src/main/java/com/barns/app/domain/usecase/catalog/GetCatalogItemDetailUseCase.kt
package com.barns.app.domain.usecase.catalog

import com.barns.app.domain.model.CatalogItem
import com.barns.app.domain.repository.CatalogRepository

/** Returns a single official catalog item by id, if it exists. */
class GetCatalogItemDetailUseCase(
    private val repository: CatalogRepository,
) {
    suspend fun execute(id: String): CatalogItem? = repository.catalogItem(id)
}

File: apps/ios/Barns/BarnsApp/Domain/UseCases/Catalog/GetCatalogItemDetailUseCase.swift
/// Returns a single official catalog item by id, if it exists.
struct GetCatalogItemDetailUseCase {
    private let repository: CatalogRepository

    init(repository: CatalogRepository) {
        self.repository = repository
    }

    func execute(id: CatalogItem.ID) async throws -> CatalogItem? {
        try await repository.catalogItem(id: id)
    }
}
